docs/vcf_files_preprocessor.md

```diff
@@ -0,0 +1,117 @@
+# VCF files preprocessor
+
+The VCF files preprocessor is used for validating the datasets such that the
+inconsistencies can be easily identified. It can be used as an individual
+validator to check the validity of the information inside the VCF files, or as a
+helper tool for [VCF to BigQuery pipeline](/README.md/#vcf_to_bq). The VCF to
+BigQuery loading process may fail for ill-defined datasets, which can be
+resolved by applying
+[conflicts resolving strategies](https://docs.google.com/document/d/1JusaPllbpXynNziNzdyOdSRTM8i7UXbNvtkbrWNMq3Q).
+This tool provides insights on what conflicts are there in the VCF files and the
+resolutions that will be applied when loading the data into BigQuery. Meanwhile,
+it also makes the manual corrections easier if the proposed resolutions are not
+satisfactory.
+
+This tool generates a report that includes three types of inconsistencies:
+* Conflicting header definitions: It is common to have different definitions
+  for the same field in different VCF files. For this type of inconsistency, the
+  report lists all conflicting definitions, the corresponding file paths (up to
+  5), and the suggested resolutions.
+* Malformed Records (Optional): There can be malformed records that cannot be
+  parsed correctly by the VCF parser. In this case, the file paths and the
+  malformed variant records are included in the report for reference.
+* Undefined headers (Optional): Occasionally, some fields may be used directly
+  without a definition in the VCF files. Those fields and the inferred
```

Comment:
nit: consider rephrasing (e.g. `The tool can infer the type for such fields and provide them in the report.`).